rising/loading/dataset.py

```diff
@@ -3,19 +3,64 @@
 import os
 import pathlib
 import logging
+import dill
 from warnings import warn
 from functools import partial
 from tqdm import tqdm
-from typing import Any, Sequence, Callable, Union, List, Hashable, Dict
+from typing import Any, Sequence, Callable, Union, List, Hashable, Dict, Iterator
 
 from torch.utils.data import Dataset as TorchDset, Subset
+from multiprocessing import cpu_count, Pool as MPPool
 from torch.multiprocessing import Pool
 from rising.loading.debug_mode import get_debug_mode
 from rising import AbstractMixin
 
 logger = logging.getLogger(__file__)
 
 
+__all__ = ["Dataset", "CacheDataset", "LazyDataset", "CacheDatasetID", "LazyDatasetID", "LazyDatasetMulReturn"]
+
+
+def dill_helper(payload: Any) -> Any:
```

Comment:
Just a question: Can we make a more general version of this and also use it in our data loader if the process start method is set to `spawn`? This would maybe enable us to use multiprocessing on GPUs

Reply:
I created a separate Issue for this https://github.com/PhoenixDL/rising/issues/22 .